db/crud.go

```diff
@@ -1385,6 +1389,22 @@ func validateRoleAccessMap(roleAccess channels.AccessMap) bool {
 	return true
 }
 
+func isAccessError(err error) bool {
+	if err.Error() == base.SyncFnErrorMissingRole {
+		return true
+	}
+	if err.Error() == base.SyncFnErrorAdminRequired {
+		return true
+	}
+	if err.Error() == base.SyncFnErrorWrongUser {
+		return true
+	}
+	if err.Error() == base.SyncFnErrorMissingChannelAccess {
+		return true
+	}
+	return false
+}
+
```

Comment:
Changing this to the following seems more straightforward, but go ahead and leave as-is if you disagree.
```
	return err.Error() == base.SyncFnErrorMissingRole ||
		err.Error() == base.SyncFnErrorAdminRequired ||
		err.Error() == base.SyncFnErrorWrongUser ||
		err.Error() == base.SyncFnErrorMissingChannelAccess
```

Reply:
How about keeping a list of all the access errors and then checking for containment in that list?  The benefit is that it will be less likely for someone to forget to update `isAccessError()` since the list is defined closer to the constants.  See latest commit.